### ISSUE TITLE
Docker hub build error

### ISSUE THREAD

**Issue by mstenta** [farmOS/farmOS#49]
Docker hub builds are currently failing with the following error:

`Build failed: The command '/bin/sh -c apt-get update && apt-get install -y bzip2 git php5-geos phpunit unzip' returned a non-zero code: 100`

Example: https://hub.docker.com/r/farmos/farmos/builds/bpleynmdmn8naaz2kx59unj/

**Comment by mstenta** [comment]
Building it locally seems to work fine.

`sudo docker build .`

**Comment by mstenta** [comment]
Stack Overflow suggests that it's an HTTPS issue: https://stackoverflow.com/questions/38002543/apt-get-update-returned-a-non-zero-code-100

But I don't know if that applies here, because as I said above: it is building for me locally. It's just failing on Docker Hub.

**Comment by mstenta** [comment]
Ok, more details in the full Docker Hub output:

```
[91mE: Package 'php5-geos' has no installation candidate
E: Package 'phpunit' has no installation candidate
[0m
Package php5-geos is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source

Package phpunit is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source
However the following packages replace it:
  phpunit-story phpunit-selenium phpunit-mock-object


Removing intermediate container 5c32f9363d93

The command '/bin/sh -c apt-get update && apt-get install -y   bzip2   git   php5-geos   phpunit   unzip' returned a non-zero code: 100
```

**Comment by mstenta** [comment]
It appears that a change was made in the underlying php5.6-apache Docker image (which farmOS is based upon), that blocks the installation of Debian PHP packages.

https://github.com/docker-library/php/pull/542

https://github.com/docker-library/php/commit/dbd5d1e03d12d6405b0f7bd82623c4410679701e#diff-7dd25658243fe99c8fbeecd29c63876e

**Comment by mstenta** [comment]
Discussion: https://github.com/docker-library/php/issues/551

**Comment by hairmare** [comment]
Replace the following two lines with `libgeos-dev \` and the php-geos parts build without any system php parts.

https://github.com/farmOS/farmOS/blob/8140f90cbf04573f8f7db1453b7207e5d3c0841b/Dockerfile#L44-L45

If you need/want phpunit in you image you can install it with composer and call it from the vendor/bin subdir.

**Comment by mstenta** [comment]
Thanks @hairmare ! I will give that a try for GEOS. Good to know!

As for phpunit - I need to look back to see when/why that was added originally. Useful to have, but I'm not sure if it's necessary to include.

I also created a pull request in the PHP Docker image repo proposing that we revert the change that was made: https://github.com/docker-library/php/pull/554

It probably broke a lot of Docker builds - not just ours. And it prevents us from building a 7.x-1.0-beta16 Docker image, which is very disappointing.

**Comment by mstenta** [comment]
Tested a local build with `libgeos-dev` instead of `php5-geos` and it seemed to work fine. Pushed to Github to trigger a build on Docker Hub... we'll see if it works: https://hub.docker.com/r/farmos/farmos/builds/bffwyxtjziyjj6kfujaqkwy/

I looked into the `phpunit` dependency, and it appears it was added at the same time as php5-geos. I removed it and it didn't affect my local test, so we might not need it. I don't remember why it was necessary originally.

**Comment by mstenta** [comment]
The change in docker-library/php is not going to be reverted, so we won't have a 7.x-1.0-beta16 tagged Docker image, unfortunately. (See https://github.com/docker-library/php/issues/551)

That shouldn't be too big of an issue, though, since the Docker instructions on farmOS.org are all geared towards a dev build anyway. I plan to change that in the future, but it won't matter for 7.x-1.0-beta16: https://github.com/farmOS/farmOS.org/issues/20

In the meantime, the 7.x-1.x Docker image is building successfully now, so that's all set. Maybe it makes sense to create a one-time tag that includes the fix, and point the "latest" Docker image at that, so at least "latest" will work (I don't want to point "latest" at 7.x-1.x because it may change frequently).

**Comment by mstenta** [comment]
OK! I tagged a special 7.x-1.0-beta16-dockerfix and pointed the "latest" Docker hub image at it. I also created a 7.x-1.0-beta16 Docker hub image and pointed it at it too, so this will actually override the original, and make it possible to have a beta16 image after all! Just took a few extra revert commits. :-)

Here are the builds:

latest: https://hub.docker.com/r/farmos/farmos/builds/bdtxcbxsgfmn2bpwsk6u8te/
7.x-1.0-beta16: https://hub.docker.com/r/farmos/farmos/builds/bfjyrwjcc6vnnqujtnjlrwq/
